### PR TITLE
P2-366 Merge Feature/schema blocks into develop

### PR DESCRIPTION
**Pull request by Xyfi**
## Summary

This PR can be summarized in the following changelog entry:

* Brings the schema blocks feature code into the monorepo codebase

## Relevant technical choices:

* this merges the initial feature branch, and enables the dev team to continue their work on structured data blocks; see [epic](https://yoast.atlassian.net/browse/P2-211)

## Test instructions

This PR can be tested by following these steps:

* this is a technical change, containing foundation code. the code itself is not used yet. So, if travis succeeds, the code is OK to merge. Tests will follow upon completion of the tasks in the epic.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  * none. unused code.

## UI changes
* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Quality assurance

* [x] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

Fixes https://yoast.atlassian.net/browse/P2-366

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -17958,7 +17958,7 @@ react-dom@16.6.3:
     prop-types "^15.6.2"
     scheduler "^0.13.4"
 
-react-dom@^16.13.0, react-dom@^16.9.0:
+react-dom@^16.9.0:
   version "16.13.0"
```
Comment (increddibelly) [inline yarn.lock:17962]: I'd expect all react versions to match up - you could end up with react-dom 16.13 + react 16.12

File: yarn.lock [hunk 2]
```diff
@@ -18329,7 +18329,7 @@ react@16.6.3:
     prop-types "^15.6.2"
     scheduler "^0.13.4"
 
-react@^16.13.0, react@^16.9.0:
+react@^16.9.0:
   version "16.13.0"
```
Comment (increddibelly) [inline yarn.lock:18333]: 16.12??

**Comment by increddibelly** [comment]
@andizer is verifying that the gui doesn't break with react 16.13
if so, we will merge and add a comment the mischievous test that requires v16.12

File: packages/schema-blocks/src/core/Definition.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+		this.separator = separator;
+		this.template = template;
+		this.instructions = instructions;
+		this.tree = tree;
+	}
+
+	/**
+	 * Returns the configuration of this BlockDefinition.
+	 *
+	 *@returns The configuration.
```
Comment (andizer) [inline packages/schema-blocks/src/core/Definition.ts:43]:
```suggestion
	 * @returns The configuration.
```

File: packages/schema-blocks/src/core/blocks/BlockDefinition.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+export interface RenderSaveProps extends BlockSaveProps<Record<string, unknown>> {
+	clientId?: string;
+}
+
+export type MutableBlockConfiguration = {
+	-readonly [K in keyof BlockConfiguration]: BlockConfiguration[K]
+}
+
+/**
+ * BlockDefinition clas
```
Comment (andizer) [inline packages/schema-blocks/src/core/blocks/BlockDefinition.ts:23]:
```suggestion
 * BlockDefinition class
```

**Review by andizer — COMMENTED** [review]
Two minor things

**Pull request merged**